### ISSUE TITLE
[Feature request] Single-file publication support

### ISSUE THREAD

**Issue by nsentinel** [curiosity-ai/rocksdb-sharp#12]
It would be nice to have support for the single-file publication (.NET 5+).

Currently, library throws the exception:

```
System.TypeInitializationException The type initializer for 'RocksDbSharp.Native' threw an exception. 
Void .ctor()
   at RocksDbSharp.OptionsHandle..ctor()
   at RocksDbSharp.ColumnFamilyOptions..ctor()
   at RocksDbSharp.DbOptions..ctor()

NotSupportedException CodeBase is not supported on assemblies loaded from a single-file bundle. 
System.String get_CodeBase()
```

Initializer here: https://github.com/curiosity-ai/rocksdb-sharp/blob/ef23652f005c60bed6c73d44d2c5b12ff2842ab6/csharp/src/AutoNativeImport.cs#L375

can use [AppContext.BaseDirectory](https://docs.microsoft.com/en-us/dotnet/api/system.appcontext.basedirectory?view=net-5.0) and wrap `CodeBase` lines to try-catch clause to prevent it.

publish command:

```
dotnet publish -c Release --runtime win-x64 -p:PublishSingleFile=true -p:DebugType=embedded --self-contained true
```

Thanks in advance.

**Comment by NSentinel** [comment]
Seems it already implemented with this PR: https://github.com/curiosity-ai/rocksdb-sharp/pull/7

**Comment by theolivenbaum** [comment]
I've merged #7 but from reading the error, I think it won't fix this issue. I'll push another change for it
